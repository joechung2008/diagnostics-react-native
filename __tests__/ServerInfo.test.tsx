import ServerInfo from '@/ServerInfo';
import { render, screen } from '@testing-library/react-native';
import React from 'react';

// Mock the ThemeContext
jest.mock('@/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f8f9fa',
      text: '#212529',
      tableBorder: '#dee2e6',
      tableHeaderBg: '#f8f9fa',
    },
  }),
}));

describe('ServerInfo', () => {
  it('renders correctly with all server info', () => {
    const props = {
      deploymentId: 'deploy-123',
      extensionSync: {
        totalSyncAllCount: 42,
      },
      hostname: 'server.example.com',
      nodeVersions: 'v18.17.0',
      serverId: 'server-456',
      uptime: 1234567890,
    };
    render(<ServerInfo {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with minimal server info', () => {
    const props = {
      deploymentId: 'deploy-123',
      extensionSync: {
        totalSyncAllCount: 0,
      },
      hostname: 'localhost',
      nodeVersions: undefined,
      serverId: 'server-456',
      uptime: undefined,
    };
    render(<ServerInfo {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
