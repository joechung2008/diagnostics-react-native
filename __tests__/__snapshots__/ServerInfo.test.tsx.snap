// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ServerInfo renders correctly with all server info 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f8f9fa",
        "borderColor": "#dee2e6",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#f8f9fa",
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Hostname
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        server.example.com
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Uptime
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        1234567890
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Server ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        server-456
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Deployment ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        deploy-123
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Node Versions
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        v18.17.0
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Extension Sync | Total Sync All Count
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        42
      </Text>
    </View>
  </View>
</View>
`;

exports[`ServerInfo renders correctly with minimal server info 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "padding": 10,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f8f9fa",
        "borderColor": "#dee2e6",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#f8f9fa",
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "#212529",
            "flex": 1,
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Hostname
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        localhost
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Uptime
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Server ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        server-456
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Deployment ID
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        deploy-123
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Node Versions
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#dee2e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        Extension Sync | Total Sync All Count
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#212529",
              "flex": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;
